src/Compilers/Core/Portable/Operations/IIndexOperation.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Represents an index operation.
+    /// <para>
+    /// Current Usage:
+    ///  (1) C# index expressions
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to change it in the future.
+    /// </remarks>
+    public interface IIndexOperation : IOperation
+    {
+        /// <summary>
+        /// The operand.
+        /// </summary>
+        IOperation Operand { get; }
+
+        /// <summary>
+        /// <code>true</code> if this is a 'lifted' index operation.  When there is an 
+        /// operator that is defined to work on a value type, 'lifted' operators are 
+        /// created to work on the <see cref="System.Nullable{T}"/> versions of those
+        /// value types.
+        /// </summary>
+        bool IsLifted { get; }
```

Comment:
>IsLifted [](start = 13, length = 8)

I don't see how an index operation could be lifted. Can you please give an C# example of how this arises?

Reply:
If you're talking about a user-declared `operator ^`, that should probably be an ordinary user-defined unary operator invocation rather than this node.

---
In reply to: [206305663](https://github.com/dotnet/roslyn/pull/28259#discussion_r206305663) [](ancestors = 206305663)